frontend/src/config/config.test.ts

```diff
@@ -0,0 +1,121 @@
+    });
+  });
+
+  it('should have translations with correct depth', () => {
+    Object.keys(configMap).forEach(country => {
+      const config = configMap[country as Country];
+      const layers = getRawLayers(country as Country);
+      const translation = getTranslation(country as Country);
+
+      // TODO - activate translation for reports and tables
+      // const reports = config.rawReports;
+      // const tables = config.rawTables;
+
+      // All the elements need translation and should appear at least once in the translations:
+      /**
+       * - All the keys in config.categories
+       * - All the key
+       */
+      let itemsToTranslate: string[] = [];
+
+      // categories
+      const categoryKeys = jp
+        .query(config.appConfig.categories, '$.*')
+        .filter(
+          (value: any) =>
+            typeof value === 'string' ||
+            (typeof value === 'object' &&
+              value !== null &&
+              !Array.isArray(value)),
+        )
+        .map((value: any) =>
+          typeof value === 'object' ? Object.keys(value) : value,
+        )
+        .flat()
+        .map((value: any) => startCase(value));
+
+      // grouped layer titles
+      const categoryGroupTitles = jp.query(
+        config.appConfig.categories,
+        '$..group_title',
+      );
+
+      // layers titles
+      const layerTitles = jp.query(layers, '$..title');
+      const layerLegendTexts = jp.query(layers, '$..legend_text');
+      const chartLegendLabels = jp.query(
+        layers,
+        '$..chart_data.fields[*].label',
+      );
+      // const legendLabels = jp.query(layers, '$..legend[*].label');
+      // eslint-disable-next-line fp/no-mutation
+      itemsToTranslate = [
+        ...Array.from(translationKeys),
+        ...categoryKeys,
+        ...categoryGroupTitles,
+        ...layerTitles,
+        ...layerLegendTexts,
+        ...chartLegendLabels,
+      ];
+      // Deduplicate items using a Set
+      // eslint-disable-next-line fp/no-mutation
+      itemsToTranslate = Array.from(new Set(itemsToTranslate));
+
+      Object.entries(translation).forEach(([key, value]) => {
+        if (key === 'en') {
+          // nothing to do, we assume keys are in english
+        } else {
+          // compare translation with itemsToTranslate
+          const missingFields: string[] = [];
+          itemsToTranslate.forEach(item => {
+            if (
+              item !== '' &&
+              !Object.prototype.hasOwnProperty.call(value, item)
+            ) {
+              // eslint-disable-next-line fp/no-mutating-methods
+              missingFields.push(item);
+            }
+          });
+          if (missingFields.length > 0) {
+            // eslint-disable-next-line no-console
+            console.log(
+              `${missingFields.length} missing fields for ${country} - language: ${key}`,
+            );
+            console.error(missingFields);
+          }
```

Comment:
I think this is a really helpful change and I'm really thankful that you put this step into existing processes.

A minor suggestion, I found the console output a bit hard to read. I think using console.table could help to read the missing fields

```suggestion
          if (missingFields.length > 0) {
            // Create an array to hold the table data
            const tableData: { [key: string]: string }[] = [];

            // Populate the table data with missing fields
            missingFields.forEach(field => {
              const row: { [key: string]: string } = {
                [`Missing Fields: ${country} - ${key}`]: `${field.slice(0, 64)}${field.length > 64 ? '...' : ''}`,
              };
              // eslint-disable-next-line fp/no-mutating-methods
              tableData.push(row);
            });

            // Print the table to the console
            // eslint-disable-next-line no-console
            console.table(tableData);
          }
```

| Before | After |
| - | - |
| <img width="522" alt="Screenshot 2024-07-11 at 4 59 45 PM" src="https://github.com/user-attachments/assets/b3444ae5-7c81-4fb5-9e86-f4b55f7e9cd6"> |  <img width="601" alt="Screenshot 2024-07-11 at 4 58 58 PM" src="https://github.com/user-attachments/assets/e66b59ea-d82a-41a9-9a98-4c3ebaaf82cf"> |

Reply:
> A minor suggestion, I found the console output a bit hard to read. I think using console.table could help to read the missing fields

I like this suggestion!